main.go

```diff
@@ -186,11 +226,183 @@ func newCreateCustomerCommand(db *sql.DB) *cli.Command {
-			w.Flush()
+			printCustomer(int(customerID), email, state)
+
+			return nil
+		},
+	}
+}
+
+func newCreateProductCommand(db *sql.DB) *cli.Command {
+	return &cli.Command{
+		Name:  "create-product",
+		Usage: "Creates a new product to go in the products database, must specify name and price",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "sku",
+				Usage: "the sku of the product",
+			},
+		},
+		Action: func(cCtx *cli.Context) error {
+			if cCtx.NArg() < 2 {
+				return errors.New("Must specify name and price")
+			}
+
+			name := cCtx.Args().Get(0)
+			price, err := strconv.ParseFloat(cCtx.Args().Get(1), 32)
+			if err != nil {
+				return err
+			}
+			sku := cCtx.String("sku")
+			args := []any{}
+
+			w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.Debug)
+			var insertStatement string
+			if sku != "" {
+				insertStatement = "INSERT INTO Products (name, price, sku) VALUES (?, ?, ?)"
+
+			} else {
+
+				insertStatement = "INSERT INTO Products (name, price) VALUES (?, ?)"
+
+			}
+			args = append(args, name, price, sku)
+
+			err = productsInsertHelper(db, args, insertStatement, w)
+			if err != nil {
+				return err
+			}
+
+			return nil
+
+		},
+	}
+}
+
+func newCreateOrderCommand(db *sql.DB) *cli.Command {
+	return &cli.Command{
+		Name:      "create-order",
+		Usage:     "Creates a new order to go in the order database, must specify customer_id and product_id",
+		ArgsUsage: "PRODUCT_ID CUSTOMER_ID",
+		Action: func(cCtx *cli.Context) error {
+			if cCtx.NArg() < 2 {
+				return errors.New("Must specify product_id and customer_id")
+			}
+			pID, err := strconv.Atoi(cCtx.Args().Get(0))
+			if err != nil {
+				return errors.New("Must be valid integer")
+			}
+			cID, err := strconv.Atoi(cCtx.Args().Get(1))
+			if err != nil {
+				return errors.New("Must be valid integer")
+			}
+			insertStatement := "INSERT INTO Orders (customer_id, product_id) VALUES (?, ?)"
+			res, err := db.Exec(insertStatement, cID, pID)
+			if err != nil {
+				return errors.New("Violates Foreign Key constraints")
+			}
+			orderID, err := res.LastInsertId()
+			if err != nil {
+				return err
+			}
+			printOrder(int(orderID), cID, pID)
+
+			return nil
+		},
+	}
+}
+
+func newShowCustomerCommand(db *sql.DB, ctx context.Context) *cli.Command {
+	return &cli.Command{
+		Name:  "show-customers",
+		Usage: "displays all the customers inside the customers database, optional flags to filter by email and state",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "email",
+				Usage: "the email of the customer",
+			},
+			&cli.StringFlag{
+				Name:  "state",
+				Usage: "the state of the customer",
+			},
+		},
+		Action: func(cCtx *cli.Context) error {
+			w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.Debug)
+			statement := "SELECT * FROM Customers WHERE Email LIKE CONCAT('%', ?, '%') AND State LIKE CONCAT('%', ?, '%')"
+			email := cCtx.String("email")
+			state := cCtx.String("state")
```

Comment:
This is actually pretty clever - the reason this works for every variation of the email/state flag being set/unset is because `CONCAT('%', ?, '%')` with an empty string get converted to a `%%`, which will match anything. 

That being said, it's unfortunately not very efficient. In the real world, where we potentially have to filter through hundreds of thousands of entries in a database (or more), we'd want to make sure not to add an extra/unnecessary filters to our database query since this can potentially dramatically reduce performance. 

No need to change that here for our purposes - just wanted to call that out.